packages/web/src/scss/components/SegmentedControl/_tools.scss

```
@@ -0,0 +1,15 @@
+@use '@tokens' as tokens;
+@use 'sass:map';
+@use 'theme';
+
+@mixin generate-item-transform($max-items) {
+    @for $i from 1 through $max-items {
+        [type='radio']:nth-last-of-type(#{$i + 1}):checked ~ .SegmentedControlItem__label:last-of-type::after,
+        .SegmentedControl
+            > :nth-last-of-type(#{$i + 1}):has([type='radio']:checked)
+            ~ :last-of-type
+            > .SegmentedControlItem__label::after {
```

Tiny optimization: maybe you could set the variable on the label itself to save some letters?

```suggestion
        [type='radio']:nth-last-of-type(#{$i + 1}):checked ~ .SegmentedControlItem__label:last-of-type,
        .SegmentedControl
            > :nth-last-of-type(#{$i + 1}):has([type='radio']:checked)
            ~ :last-of-type
            > .SegmentedControlItem__label {
```

It's up to you, nothing major.